opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -99,6 +100,46 @@ def get_context(self) -> 'SpanContext':
         """
 
 
+class TraceOptions(int):
+    """A bitmask that represents options specific to the trace.
+
+    The only supported option is the "recorded" flag (``0x01``). If set, this
+    flag indicates that the trace may have been recorded upstream.
+
+    See the `W3C Trace Context`_ spec for details.
+
+    .. _W3C Trace Context: https://www.w3.org/TR/trace-context/#trace-flags
+    """
+    DEFAULT = 0x00
+    RECORDED = 0x01
+
+    @classmethod
+    def get_default(cls) -> 'TraceOptions':
+        return cls(cls.DEFAULT)
+
+
+DEFAULT_TRACEOPTIONS = TraceOptions.get_default()
+
+
+class TraceState(typing.Dict[str, str]):
+    """A list of key-value pairs that carries system-specific config.
+
+    Keys are strings of up to 256 characters containing only lowercase letters
```

Comment:
Similar like `TraceOptions`, it'll be better to put a reference link to the W3C spec.
BTW, do we intend to describe the syntax rules here, or a link to the spec would be sufficient?

Reply:
I think we need the syntax too unless we're going to be stricter about following the spec.

Reply:
I thought we're targeting 100% compliant with the spec, aren't we?

Reply:
It's clear what it means for the propagator to be 100% compliant, but I wasn't sure we could say the same thing for`SpanContext` since we're not e.g. checking that these strings are valid in the API.

Looking at this again I think you're right: we can just link to the spec and leave it up to the implementation to check that the fields are valid.